### DISCUSSION TITLE
Some question about executor and golang

### DISCUSSION THREAD

**Discussion by kysshsy** [go-co-op/gocron#252] [Q&A] [answered]
I found 
https://github.com/go-co-op/gocron/blob/560e7b9d9a01ee470592922a80761b68cf850870/executor.go#L110
it sends a value to buffered channel 
https://github.com/go-co-op/gocron/blob/560e7b9d9a01ee470592922a80761b68cf850870/executor.go#L111
next line, it read from the same buffered channel.
Is it possible that it reads the value it sends and the executor doesn't stop.
Is this a golang skill(trick) ?

excuse my poor english

**Comment by JohnRoesler** [comment, marked as answer]
The write on line 110 is ready by the channel read on line 100: https://github.com/go-co-op/gocron/blob/560e7b9d9a01ee470592922a80761b68cf850870/executor.go#L100-L105 

To my knowledge it couldn't read on line 111 before the 110 write was ready by 100. 111 then waits for the write on 103. 

We'd have to dig deeper into how channels / goroutines work to understand if its possible if say one of the gorouintes is running much slower and that causes some out of sync issues.

**Reply by kysshsy** [reply]
thanks for your help
